module1/tasks.js

```diff
@@ -0,0 +1,189 @@
+  it('should accept a string type only', () => {
+    expect(() => alphabetSort()).to.throw();
+    expect(() => alphabetSort('Text')).not.to.throw();
+  });
+
+  it('should convert the letters of a given string in alphabetical order', () => {
+    expect(alphabetSort('Python')).to.equal('Phnoty');
+  });
+});
+
+// ==== Get min integer ====
+const getSecondMinimum = (arr) => {return arr.sort()[1]};
+
+describe('getSecondMinimum', () => {
+  it('should get array of integers and return second minimum value', () => {
+    expect(getSecondMinimum([5,0,7,3,8])).to.equal(3);
+  });
+});
+
+// ==== Double every even integer ====
+const doubleEveryEven = (arr) => { 
+  return arr.map(e => e % 2 == 0 ? e*2 : e);
+};
+
+describe('doubleEveryEven', () => {
+  it('should get array of integers and return another array of integers where every even number is doubled', () => {
+    expect(doubleEveryEven([2,0,7,3,8,4])).to.eql([4,0,7,3,16,8]);
+  });
+});
+
+// ==== Create array with all possible pairs of two arrays ====
+const getArrayElementsPairs = (a1, a2) => {
+  let result = new Array();
+  for(let e1 of a1) {
+    for(let e2 of a2) {
+      result.push([e1, e2]);
+    }
+  }
+  return result;
+};
+
+describe('getArrayElementsPairs', () => {
+  it('should get two arrays and return array containing each possible pair from the arrays', () => {
+    expect(getArrayElementsPairs([1,2], ['a', 'b'])).to.eql([[1, "a"], [1, "b"], [2, "a"], [2, "b"]]);
+  });
+});
+
+// ==== Deep equal ====
+const deepEqual = (a1 , a2) => {  
+  if(Object.entries(a1).length != Object.entries(a2).length) {
+    //console.log("lengths are not equal " + Object.entries(a1).length + "  "  + Object.entries(a2).length);
+    return false;
+  }  
+  for (let [key, value] of Object.entries(a1)) {
+    //console.log("iterate of " + key + " ; " + value );
+    if(typeof value === 'object' && value !== null) {
+      //console.log("inside comparing objects " + value + " ; " + a2[key]);      
+      let innerObjTheSame = deepEqual(value, a2[key]);
+      //console.log("result of comparing " + innerObjTheSame);
+      if(!innerObjTheSame) {
+        //console.log("returning false because of " + innerObjTheSame);
+        return false;
+      }
+    }
+    else if(value !== a2[key]) {
+      //console.log("returning false because of value !== a2[key]   " + value + " ; " + a2[key]);
+      return false;    
+    }  
+  }
+  return true;
+};
+
+describe('deepEqual', () => {
+  let obj = {here: {is: "an"}, object: 2};
+  it('should get two values and returns true only if they are the same value or are objects with the same properties, where the values of the properties are equal', () => {
+    expect(deepEqual(obj, obj)).to.eql(true);    
+    expect(deepEqual(obj, {here: 1, object: 2})).to.eql(false);
+    expect(deepEqual(obj,  {here: {is: "an"}, object: 2})).to.eql(true);
+  }); 
+});
+
+function isString(input) {
+  return Object.prototype.toString.call(input) === "[object String]";
```

Comment:
I don't agree with the solution. It works but you solve wrong task.
I will give you a hint. What is a string in JS, in Java, in any of programming languages?